### DISCUSSION TITLE
How to add policy for view in sql editor

### DISCUSSION THREAD

**Discussion by xuanvan229** [supabase/supabase#12531] [Questions] [answered]
I've used this query successfully before, but it seems to be incorrect now and I'm not sure why.

<img width="681" alt="Screenshot 2023-02-19 at 16 17 35" src="https://user-images.githubusercontent.com/22408713/219939519-7e611a38-f34b-4161-abd3-6097414f4584.png">

**Comment by GaryAustin1** [comment]
Probably caught up in this https://github.com/supabase/supabase/discussions/9314
If you don't see something close to your issue in that you might ask (especially if your SQL current_user is postgres (versus the old supabase_admin)

**Reply by xuanvan229** [reply]
Thanks. @GaryAustin1 
But I have tried to grant `authenticated`  to`postgres`. but still not work

<img width="812" alt="Screenshot 2023-02-20 at 07 57 58" src="https://user-images.githubusercontent.com/22408713/219986933-7d54e948-a57d-4937-b5fb-6e9f627c7d53.png">

**Comment by steve-chavez** [comment, marked as answer]
@xuanvan229 On PostgreSQL 15 you no longer need changing the `owner` for views to work with RLS. Instead you can do:

```sql
CREATE VIEW sorted_titles WITH (security_invoker) as SELECT title FROM posts ORDER BY title;
```

Also see https://github.com/supabase/supabase/discussions/901

**Reply by xuanvan229** [reply]
@steve-chavez  Thank you so much 🙏
